src/libs/PolicyUtils.ts

```diff
@@ -0,0 +1,217 @@
+import {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import Str from 'expensify-common/lib/str';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+import {PersonalDetails, Policy, PolicyMembers, PolicyTags} from '../types/onyx';
+
+type MemberEmailsToAccountIDs = Record<string, number>;
+type PersonalDetailsList = Record<string, PersonalDetails>;
+type UnitRate = {rate: number};
+
+/**
+ * Filter out the active policies, which will exclude policies with pending deletion
+ * These are policies that we can use to create reports with in NewDot.
+ */
+function getActivePolicies(policies: OnyxCollection<Policy>): Policy[] | undefined {
+    return Object.values(policies ?? {}).filter<Policy>(
+        (policy): policy is Policy =>
+            policy !== null && policy && (policy.isPolicyExpenseChatEnabled || policy.areChatRoomsEnabled) && policy.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+    );
+}
+
+/**
+ * Checks if we have any errors stored within the POLICY_MEMBERS. Determines whether we should show a red brick road error or not.
+ * Data structure: {accountID: {role:'user', errors: []}, accountID2: {role:'admin', errors: [{1231312313: 'Unable to do X'}]}, ...}
+ */
+function hasPolicyMemberError(policyMembers: OnyxEntry<PolicyMembers>): boolean {
+    return Object.values(policyMembers ?? {}).some((member) => Object.keys(member?.errors ?? {}).length > 0);
+}
+
+/**
+ * Check if the policy has any error fields.
+ */
+function hasPolicyErrorFields(policy: OnyxEntry<Policy>): boolean {
+    return Object.keys(policy?.errorFields ?? {}).some((fieldErrors) => Object.keys(fieldErrors ?? {}).length > 0);
+}
+
+/**
+ * Check if the policy has any errors, and if it doesn't, then check if it has any error fields.
+ */
+function hasPolicyError(policy: OnyxEntry<Policy>): boolean {
+    return Object.keys(policy?.errors ?? {}).length > 0 ? true : hasPolicyErrorFields(policy);
+}
+
+/**
+ * Checks if we have any errors stored within the policy custom units.
+ */
+function hasCustomUnitsError(policy: OnyxEntry<Policy>): boolean {
+    return Object.keys(policy?.customUnits?.errors ?? {}).length > 0;
+}
+
+function getNumericValue(value: number, toLocaleDigit: (arg: string) => string): number | string {
```

Comment:
Is the return type should be only a number?

Reply:
Hi, `numValue.toFixed()` returns string, so it's necessary. Also I have renamed PolicyUtils.js to PolicyUtils.ts, so that's solved - must've added it when solving some of the merge conflicts